src/main/scala/ai/privado/exporter/SourceExporter.scala

```diff
@@ -83,10 +84,13 @@ class SourceExporter(
               .distinctBy(_.lineNumber)
               .distinctBy(Utilities.getFileNameForNode)
         }
-        // list of first node of every dataflow
-        val dataflowSourceList = dataFlowCache.getDataflowAfterDedup
+        // list first element of every dataflow
+        val dataflowSourceList = dataflowsOutput
+          .flatMap(_._2)
           .filter(_.sourceId.equals(entrySet._1))
-          .map(pathModel => dataflows(pathModel.pathId).elements.head)
+          .flatMap(_.sinks)
+          .flatMap(_.paths)
+          .map(path => dataflows(path.pathId).elements.head)
```

Comment:
why are we picking path.path and getting info from dataflows, when `path.ocurrences.head` will give us the first element ?